verifier/src/quote_enclave_identity.rs

```diff
@@ -107,6 +111,46 @@ pub struct SignedQeIdentity<'a> {
     signature: Vec<u8>,
 }
 
+impl<'a> SignedQeIdentity<'a> {
+    /// Verify the `enclaveIdentity` signature and time are valid.
+    ///
+    /// # Arguments
+    /// - `key` - The public key to verify the `enclaveIdentity` signature
+    /// - `time` - The current system time
+    ///   This is expected to be generated by:
+    ///     ```ignore
+    ///     let time = DateTime::from_system_time(SystemTime::now()).unwrap();
+    ///     ```
+    ///   or equivalent
+    pub fn verify(self, key: &VerifyingKey, time: DateTime) -> Result<(), Error> {
+        self.verify_signature(key)?;
+        self.verify_time(time)?;
+        Ok(())
+    }
+
+    fn verify_signature(&self, key: &VerifyingKey) -> Result<(), Error> {
+        let qe_identity = self.enclave_identity.get();
+        let signature =
+            Signature::try_from(&self.signature[..]).map_err(|_| Error::SignatureDecodeError)?;
+        key.verify(qe_identity.as_bytes(), &signature)
+            .map_err(|_| Error::SignatureVerification)?;
+        Ok(())
+    }
+
+    fn verify_time(&self, time: DateTime) -> Result<(), Error> {
+        let qe_identity = QeIdentity::try_from(self)?;
+        let issue_date = qe_identity.issue_date.parse::<DateTime>()?;
+        let next_update = qe_identity.next_update.parse::<DateTime>()?;
+        if time < issue_date {
+            Err(Error::QeIdentityNotYetValid)
+        } else if time >= next_update {
+            Err(Error::QeIdentityExpired)
```

Comment:
I didn't think about it until writing up the SignedJson comment above, but the NotYetValid and Expired are the only errors that are TCB vs QEIdentity.  The error message is "`The QE identity signature could not be verified: {error}`" so especially if the SignedJson approach is more desirable there should be no harm in genericising these errors.